### PR TITLE
Merge ExistingConnectionTest with ConnectionSpecificationTest

### PR DESCRIPTION
**Pull request by AndriySvyryd**
And some formatting fixes

### CODE REVIEW

**Comment by AndriySvyryd** [comment]
@Pilchie FYI

File: src/EFCore/Metadata/Conventions/ForeignKeyPropertyDiscoveryConvention.cs [hunk 1]
```diff
@@ -39,20 +39,20 @@ namespace Microsoft.EntityFrameworkCore.Metadata.Conventions
...
+        INavigationAddedConvention,
         ISkipNavigationForeignKeyChangedConvention,
         ISkipNavigationInverseChangedConvention,
+        IPropertyAddedConvention,
+        IPropertyNullabilityChangedConvention,
+        IPropertyFieldChangedConvention,
         IKeyAddedConvention,
         IKeyRemovedConvention,
         IEntityTypePrimaryKeyChangedConvention,
+        IEntityTypeMemberIgnoredConvention,
```
Comment (smitpatel) [inline src/EFCore/Metadata/Conventions/ForeignKeyPropertyDiscoveryConvention.cs:55]: What order is this? Alphabetical could help.

File: test/EFCore.SqlServer.FunctionalTests/ConnectionSpecificationTest.cs [hunk 2]
```diff
@@ -450,5 +452,100 @@ private class Customer
             public string Fax { get; set; }
             // ReSharper restore UnusedMember.Local
         }
+
+        // See aspnet/Data#135
```
Comment (smitpatel) [inline test/EFCore.SqlServer.FunctionalTests/ConnectionSpecificationTest.cs:456]: I really wonder what this issue refers to. But I don't know if I want to find out.

File: test/EFCore.SqlServer.FunctionalTests/ConnectionSpecificationTest.cs [hunk 3]
```diff
@@ -450,5 +452,100 @@ private class Customer
...
+            return Can_use_an_existing_closed_connection_test(openConnection: false);
+        }
+
+        [ConditionalFact]
+        public Task Can_use_an_existing_open_connection()
+        {
+            return Can_use_an_existing_closed_connection_test(openConnection: true);
+        }
+
+        private static async Task Can_use_an_existing_closed_connection_test(bool openConnection)
```
Comment (smitpatel) [inline test/EFCore.SqlServer.FunctionalTests/ConnectionSpecificationTest.cs:469]: Just make this public and conditional theory?

File: src/EFCore/Metadata/Conventions/ForeignKeyPropertyDiscoveryConvention.cs (hunk 1 above)
Comment (AndriySvyryd) [reply src/EFCore/Metadata/Conventions/ForeignKeyPropertyDiscoveryConvention.cs:55]: It's relatedness order

**Review by smitpatel — APPROVED** [review]

**Pull request merged**